task/upload.go

```diff
@@ -573,6 +631,38 @@ func uploadTaskOutputLocations(tctx *TaskContext) ([]OutputName, []clients.Outpu
 	return outputNames, outputLocations, nil
 }
 
+func clipTaskOutputLocations(tctx *TaskContext) ([]OutputName, []clients.OutputLocation, error) {
+	playbackId := tctx.OutputAsset.PlaybackID
+	outURL := tctx.OutputOSObj.URL
+	sourceURL := tctx.Task.Params.Clip.URL
+	cleanUrl := strings.TrimSuffix(sourceURL, "/output.m3u8")
+
+	clipURL, err := url.Parse(cleanUrl)
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("error parsing clip URL: %w", err)
+	}
+
+	outputNames, outputLocations, err := outputLocations(outURL, OUTPUT_ENABLED, playbackId, OUTPUT_ENABLED, playbackId, "", "", false)
+
+	if err != nil {
+		return nil, nil, err
+	}
+	clipOutputLocationUrl := clipURL.JoinPath("/clip_" + playbackId).String()
```

Comment:
Won't this make the clips ephemeral? Why aren't clips stored in a similar pattern as any other asset? i.e. on the VOD object store & with a path like `/hls/<playback_id>/...` (the asset playback ID, not the stream)

Being out of context, it seems like clips are just a special kind of upload with start/end timestamps. The outputs should be just the same tho and go to the same place as regular VODs. We also don't need to make the files "related" on the object store, like storing clips inside the stream folder, just need them to be related in the API (which they already are, through `source.sessionId`). I'm probably missing something though.

Reply:
Yes - this is the temporary location for the processing clip that I was referring on the studio PR - the actual final HLS and MP4 coming out of the vod job will be in the normal path for assets